src/main/java/Network.java

```diff
@@ -17,12 +20,19 @@ public void start() {
     }
 
     private void performLearning() {
-        final int eachIteration = 50;
-        network.addListeners(new ScoreIterationListener(eachIteration));
+        CollectScoresIterationListener sc= new CollectScoresIterationListener();
```

Comment:
Please change the name of this variable. Having names like "sc" in the code make it unreadable long-term.